services/venice-server/src/main/java/com/linkedin/venice/listener/OutboundHttpWrapperHandler.java

```diff
@@ -168,7 +171,12 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
       statsHandler.setResponseStatus(responseStatus);
     }
 
-    FullHttpResponse response = new DefaultFullHttpResponse(HTTP_1_1, responseStatus, body);
+    if (response != null) {
+      ctx.writeAndFlush(response);
```

Comment:
Since just one branch returns early it might be good to have early return in that branch itself?

Reply:
This change prevents the race condition too as it ensures that `statsHandler.setResponseStatus(responseStatus)` will be invoked before the handlers get invoked